### PR TITLE
Specify odd number of master availability zones (otherwise kops barfs)

### PR DESCRIPTION
**Pull request by edahlseng**

### CODE REVIEW

**Comment by FutureSharks** [comment]
I like it! I guess the logic is be like this? Region with even number of AZs should have `(number of AZs - 1)` masters. Correct?

e.g.

2x AZs (e.g. Mumbai, London) = 1 masters
6x AZs (N. Virginia) = 5 masters

File: module/data.tf [hunk 1]
```diff
@@ -37,7 +37,11 @@ locals {
   # Number master resources to create. Defaults to the number of AZs in the region but should be 1 for regions with odd number of AZs.
   master_resource_count = "${var.force_single_master == 1 ? 1 : length(local.az_names)}"
   # Master AZs is used in the `kops create cluster` command
-  master_azs = "${var.force_single_master == 1 ? element(local.az_names, 0) : join(",", local.az_names)}"
+  odd_number_of_azs = "${length(local.az_names) / 2 == ceil(length(local.az_names) / 2)}"
```
Comment (FutureSharks) [inline module/data.tf]: Won't `odd_number_of_azs` be 0 when AZ count is 3?

`length(local.az_names) / 2 == ceil(length(local.az_names) / 2)`
is `3/2 == ceil(3 / 2)`
is `1.5 == ceil(1.5)`
is `1.5 == 2`
is `False` or `0`

Or am I wrong?

Comment (edahlseng) [reply module/data.tf]: Uff, yes. Embarrassing typo, will fix.

**Comment by edahlseng** [comment]
@FutureSharks yep, that's the logic I was going for!

**Comment by edahlseng** [comment]
I just tried running this on a cluster locally, and it looks like kops is correctly called with an odd number of availability zones, but a master node is still created in every availability zone. Digging in...

**Comment by edahlseng** [comment]
Okay, just pushed a fix that I believe should properly create the right number of resources.

**Comment by FutureSharks** [comment]
> I believe should properly create the right number of resources.

Did you test it in a region with 2x AZs and one with 3x AZs?

**Comment by edahlseng** [comment]
I have not, I've only tested in a region with 6x AZs. I probably won't have a chance today to get to any more testing, but can explore this weekend if you aren't able to get to it either.

**Comment by FutureSharks** [comment]
Cool. I will have time to test it mid-early Jan. This was on my todo list anyway:
https://github.com/FutureSharks/tf-kops-cluster/blob/master/TODO.md

I think it would also be important to be able to set master count manually because I don't think anyone will want to run 6x masters in `us-east-1` region, that's too many.

**Comment by edahlseng** [comment]
@FutureSharks do you want to include the changes in here or wrap this up with the ability to set master count manually?

**Comment by FutureSharks** [comment]
Let's keep it separate.

Is this PR ready to test?

**Comment by edahlseng** [comment]
Yep, this PR is ready to test. I've pulled in your latest changes, and confirmed that it works in us-east-1 region.

**Comment by FutureSharks** [comment]
Great! I'll check it out next week.

**Comment by FutureSharks** [comment]
Hi @edahlseng ,

When I test this with the terraform code in `/example` it fails with this error when creating the cluster with kops:

```
* module.cluster2.null_resource.create_cluster: 1 error(s) occurred:

* Error running command 'kops create cluster --cloud=aws --dns private --authorization AlwaysAllow --networking flannel --zones=eu-west-1a,eu-west-1b,eu-west-1c --node-count=1 --master-zones=eu-west-1a,eu-west-1b --target=terraform --api-loadbalancer-type=public --vpc=vpc-29f0244f --state=s3://kops-state-store-5782238 --kubernetes-version 1.7.10 cluster2.my-public-domain.com': exit status 1. Output: I0206 14:46:31.043160   33182 create_cluster.go:971] Using SSH public key: /Users/max.williams/.ssh/id_rsa.pub
I0206 14:46:32.470639   33182 subnets.go:184] Assigned CIDR 172.20.32.0/19 to subnet eu-west-1a
I0206 14:46:32.470659   33182 subnets.go:184] Assigned CIDR 172.20.64.0/19 to subnet eu-west-1b
I0206 14:46:32.470665   33182 subnets.go:184] Assigned CIDR 172.20.96.0/19 to subnet eu-west-1c

Spec.EtcdClusters[0].Members: Invalid value: 2: Should be an odd number of master-zones for quorum. Use --zones and --master-zones to declare node zones and master zones separately
```

This should work by default. Can you fix it?

**Comment by edahlseng** [comment]
@FutureSharks, I'm sorry I missed your comments on this PR. Unfortunately the team that I'm working with has moved our infrastructure, and so we are no longer using this package. Given that, I no longer have the bandwidth to support this PR, and so I'm closing it.

**Pull request closed without merging**